Front-end updated with license detection

kawsarahamad opened the pull request.

Shahrier-Al-Tanzim committed:
Front-end updated with license detection

kawsarahamad commented on src/controllers/user/user.controllers.js:
```diff
@@ -15,20 +22,50 @@ const getUserDashboard = async (req, res) => {
   }
 }
 
+// const addVehicle = async (req, res) => {
```
Remove your comments

Shahrier-Al-Tanzim replied:
```Okay done```

Talha76 replied:
Comments not removed yet @Ryuzaki-L-07

Talha76 commented on src/controllers/user/user.controllers.js:
```diff
@@ -15,20 +22,50 @@ const getUserDashboard = async (req, res) => {
   try {
+    const vehicle = await Vehicle.findVehicleByLicenseNumber(licenseNumber);
+    if (vehicle != null) {
+      res.render('user/user.dashboard.ejs', {
```
Does this flash work?

Talha76 commented on src/controllers/user/user.controllers.js:
```diff
@@ -15,20 +22,50 @@ const getUserDashboard = async (req, res) => {
+    const vehicle = await Vehicle.findVehicleByLicenseNumber(licenseNumber);
+    if (vehicle != null) {
+      res.render('user/user.dashboard.ejs', {
+        error: 'Vehicle already exists', // Set an error message
```
Does this way of flashing work?

Shahrier-Al-Tanzim replied:
Co-pilot gave this to me, so I think it should work, but it is not being detected in our code, so I removed it

Talha76 commented on src/routes/user/user.routes.js:
```diff
@@ -5,6 +5,6 @@ import userMiddlewares from '../../middlewares/user/user.middlewares.js';
 
 router.get('/dashboard', userMiddlewares.isloggedIn, userMiddlewares.emailVerificationMiddleware, userControllers.getUserDashboard);
-router.post('/dashboard', userMiddlewares.isloggedIn, userControllers.addVehicle);
+router.post('/dashboard/addVehicle', userMiddlewares.isloggedIn, userControllers.addVehicle);
```
Don't use upper-casing in url-path (except tokens and others)

Shahrier-Al-Tanzim replied:
so which one is preferable?
/add-vehicle or /addvehicle

Talha76 commented on src/views/user/user.dashboard.ejs:
```diff
@@ -126,5 +123,17 @@
 <script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
 <script src="extract-lptext.js"></script>
 
+<!-- Your existing HTML content -->
```
Explain this part

Shahrier-Al-Tanzim replied:
``` This is an unnecessary comment so I removed it ```

Talha76 reviewed (changes requested):
Change the requested parts.

Shahrier-Al-Tanzim committed:
Code refactored

Shahrier-Al-Tanzim committed:
Oauth Updated with Front-end

There is still one little issue with the addvehicle method, which will be implemented in the next iteration

Shahrier-Al-Tanzim committed:
Update user.controllers.js

Talha76 commented on src/routes/user/user.auth.routes.js:
```diff
@@ -1,24 +1,23 @@
-    next();
-  }
-}
+// This is the extra layer to check if the user is already authenticated
```
Why unnecessary commented code?

Talha76 committed:
Merge branch 'development' into front-end
